lieer/local.py

```diff
@@ -359,15 +352,9 @@ def load_repository (self, block = False):
 
     self.__load_cache__ ()
 
-    # load notmuch config
-    cfg = os.environ.get('NOTMUCH_CONFIG', os.path.expanduser('~/.notmuch-config'))
```

Comment:
Do we still support other configuration locations? Is this handled by notmuch?

Reply:
Notmuch now has a more sophisticated configuration handling mechanism.  It allows for multiple profiles and XDG standardised locations.  It would seem much better to allow notmuch to handle this as the user is very likely to have this set up already and it wouldn't make sense to re-implement it here.

Reply:
Ok, are you sure this is is handled automatically with this method?

Reply:
What happens if a user has multiple notmuch databases, will notmuch figure out which config is present dependent on the dir that lieer is run from? or is it handled by env variables like before but notmuch does it?

Reply:
It would seem to be dependant on the environment.  Presumably, as per what the code does here, other users would have to set the environment if they had multiple databases.  Therefore, I'm not sure if this change harms backwards compatibility.

The function calls into the notmuch library which does the configuration lookup.  The lookup is performed as per the manpage:
```
   CONFIGURATION
       Notmuch configuration file search order:

       1. File specified by notmuch --config global option; see notmuch(1).

       2. File specified by NOTMUCH_CONFIG environment variable.

       3. $XDG_CONFIG_HOME/notmuch/<profile>/config where <profile> is defined
          by   NOTMUCH_PROFILE   environment   variable   if   set,  $XDG_CON‐
          FIG_HOME/notmuch/default/config otherwise.

       4. $HOME/.notmuch-config.<profile> where <profile> is defined  by  NOT‐
          MUCH_PROFILE environment variable if set, $HOME/.notmuch-config oth‐
          erwise.
```

Maybe in the future the lieer configuration could specify a notmuch profile to use, if the user doesn't want the default.

Reply:
I think this is fine. This matches the current behavior plus new options by notmuch.